### ISSUE TITLE
circular dependencies on module collections yields without an error or warning

### ISSUE THREAD

**Issue by Ductayp** [Sleitnick/Knit#22]
I was under the impression that exposing modules to knit using the recommended method as such would help us avoid circular dependency issues beween modules that require each other. I believe Aero solved this problem.

recommended example:
Knit.MyModules = ReplicatedStorage.MyModules

However, when I have two modules that require each other, Knit yields and does not show any error or warning, making an old Roblox gotcha even harder to track down than before.

It would be nice if these circular dependencies were not an issue, such as with Nevermore and in the case that's not possible, a warning or error of some sort to help us track down the issue.

**Comment by Ductayp** [comment]
YES! I agree and I did redesign quickly but I feel like I got lucky to find the issue because the way Knit is, there was no warning or error at all. 

It can happen sometimes accidentally when we have a lot of modules and are just coding quickly, forgetting our architecture. Sometimes this happens to me when I'm trying to "rapid prototype".

Would you consider adding a warning when the code yields because of this issue?

**Comment by OverHash** [comment]
roblox-ts has a circular dependency manager which is very useful.

https://github.com/roblox-ts/roblox-ts/blob/ee09d69cca872b75871e0659724fc6471c1ad744/lib/RuntimeLib.lua#L90-L143
